### PR TITLE
ScanSummary: Make use of Jackson's syntactic sugar for Kotlin

### PR DESCRIPTION
**Pull request by sschuberth**
Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

File: model/src/main/kotlin/ScanSummary.kt [hunk 1]
```diff
@@ -94,17 +92,15 @@ class ScanSummaryDeserializer : StdDeserializer<ScanSummary>(OrtIssue::class.jav
             null
         }
 
-    private inline fun <reified T : Any> JsonNode.readValues(property: String, kClass: KClass<T>): List<T> =
-        this[property]?.map {
-            jsonMapper.treeToValue(it, kClass.java)
-        }.orEmpty()
+    private inline fun <reified T : Any> JsonNode.readValues(property: String): List<T> =
+        this[property]?.mapNotNull { jsonMapper.treeToValue<T>(it) }.orEmpty()
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/ScanSummary.kt:96]: Why did you change this to `mapNotNull`, can there be null values?

Comment (sschuberth) [reply model/src/main/kotlin/ScanSummary.kt:96]: Yes, the function signature is

    inline fun <reified T> ObjectMapper.treeToValue(n: TreeNode): T? = treeToValue(n, T::class.java)

i.e. it has a nullable return type.

Comment (mnonnenmacher) [reply model/src/main/kotlin/ScanSummary.kt:96]: Thanks.
